src/core/locator/qgslocator.h

```diff
@@ -91,14 +96,44 @@ class CORE_EXPORT QgsLocator : public QObject
      * Returns the list of filters registered in the locator.
      * \see prefixedFilters()
      */
-    QList< QgsLocatorFilter *> filters();
+    QList< QgsLocatorFilter *> filters( const QString &prefix = QString() );
+
+    /**
+     * Returns the list of prefixes in use.
+     * A prefix can have several filters
+     * \see filter()
+     * \since QGIS 3.2
+     */
+    QStringList prefixes();
+
+#ifndef SIP_RUN
 
     /**
      * Returns a map of prefix to filter, for all registered filters
      * with valid prefixes.
      * \see filters()
      */
-    QMap< QString, QgsLocatorFilter *> prefixedFilters() const;
+    QMultiMap<QString, QgsLocatorFilter *> prefixedFilters() const;
+#else
+
+    /**
+     * Returns a map of prefix to filter, for all registered filters
+     * with valid prefixes.
+     * \deprecated since QGIS 3.2 use prefixes() and filters( const QString &prefix )
+     * \see filters()
+     */
+    Q_DECL_DEPRECATED QMap<QString, QgsLocatorFilter *> prefixedFilters() const;
```

Comment:
@nyalldawson I am not sure what's the best approach here:
- implement a new conversion for multimap in conversions.sip
- return a QMap both in c++ and python (QMultiMap inherits QMap)
- remove `prefixedFilters` from Python bindings (my current approach)
- add `customPrefix` and `setCustomPrefix` methods to `QgsLocatorFilter` and change `QgsLocator::prefixedFilters` to return a QList rather than a QMap (API break) and one would call `QgsLocatorFilter::customPrefix`

Reply:
If it's not required for the unit tests, I'd be inclined to just hide the multimap methods from the bindings. Conversion code is unpleasant to write at best!